### PR TITLE
Updated keycloak to use Graal JDK8  and use the linux os images from container registry

### PR DESCRIPTION
**Pull request by sandeepbangera**
Verified locally that the server came up,and the JDK used within the container was Graal JDK.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,32 +1,76 @@
...
-    && yum-config-manager --save --setopt=ol7_ociyum_config.skip_if_unavailable=true \
-    && yum-config-manager --add-repo http://yum.oracle.com/repo/OracleLinux/OL7/developer/golang114/x86_64 \
-    && yum -y install java-11-openjdk-devel \
-    && yum -y install which \
-    && yum clean all
+FROM container-registry.oracle.com/os/oraclelinux:7-slim as builder
+
+ENV LANG=en_US.UTF-8
+
+ENV GRAALVM_PKG=https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-20.1.0/graalvm-ce-java8-linux-amd64-20.1.0.tar.gz \
```
Comment (watsh-rajneesh) [inline Dockerfile:6]: I think we cannot checkin the binary for GraalVM as it requires user to accept the license when downloading. Please see https://github.com/verrazzano/elasticsearch/pull/1/files on what we are doing for elasticsearch. We need to document where user can download the GraalVM from in the readme file and pass it as build-arg to docker build command.

File: Dockerfile [hunk 2]
```diff
@@ -1,32 +1,76 @@
...
+    PYTHONPATH=/opt/rh/llvm-toolset-7/root/usr/lib/python2.7/site-packages \
+    X_SCLS=llvm-toolset-7
+
+RUN yum update -y && \
+    yum-config-manager --save --setopt=ol7_ociyum_config.skip_if_unavailable=true && \
+    yum-config-manager --add-repo http://yum.oracle.com/repo/OracleLinux/OL7/developer/golang114/x86_64 && \
+    yum install -y bzip2-devel ed gcc gcc-c++ gcc-gfortran gzip file fontconfig less libcurl-devel make openssl openssl-devel readline-devel tar vi which xz-devel zlib-devel && \
+    yum clean all
+
+RUN fc-cache -f -v
```
Comment (watsh-rajneesh) [inline Dockerfile:21]: Can this RUN be merged with the previous RUN statement?

File: Dockerfile [hunk 3]
```diff
@@ -1,32 +1,76 @@
...
+    # Set alternative links
+    && mkdir -p "/usr/java" \
+    && ln -sfT "$JAVA_HOME" /usr/java/default \
+    && ln -sfT "$JAVA_HOME" /usr/java/latest \
+    && for bin in "$JAVA_HOME/bin/"*; do \
+        base="$(basename "$bin")"; \
+        [ ! -e "/usr/bin/$base" ]; \
+        alternatives --install "/usr/bin/$base" "$base" "$bin" 20000; \
+    done \
+    && chmod +x /usr/local/bin/gu
```
Comment (watsh-rajneesh) [inline Dockerfile:40]: Please merge the RUN statements into one.

File: verrazzano/gu-wrapper.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+#
+# LICENSE UPL 1.0
+#
+# Copyright (c) 2015 Oracle and/or its affiliates. All rights reserved.
```
Comment (watsh-rajneesh) [inline verrazzano/gu-wrapper.sh:5]: If this is a new file, the year should be 2020 right?

File: Dockerfile (hunk 1 above)
Comment (markxnelson) [reply Dockerfile:6]: you musts use graalvm ee 20.1.0.1 from OTN, not from any other source, so that the correct licenses are included

Comment (sandeepbangera) [reply Dockerfile]: Fixed the same

File: Dockerfile [hunk 5]
```diff
@@ -1,41 +1,82 @@
...
+ARG GRAALVM_BINARY="${GRAALVM_BINARY:-graalvm-ee-java11-linux-amd64-20.1.0.1.tar.gz}"
+ENV GRAALVM_DOWNLOAD_SHA256=870e51d13e7f42df50097110b14715e765e2a726aa2609c099872995c4409d8f
+
+COPY $GRAALVM_BINARY /tmp/graalvm.tar.gz
+
+RUN set -eux \
+    && echo "Checking GraalVM hash"  \
+    && echo "${GRAALVM_DOWNLOAD_SHA256} /tmp/graalvm.tar.gz" | sha256sum --check -  \
+    && tar xzf /tmp/graalvm.tar.gz -C /opt \
+    && mv /opt/graalvm-ee-* /opt/graalvm-ee \
```
Comment (watsh-rajneesh) [inline Dockerfile:31]: I think once we are done unpacking the tar.gz archive we can delete the archive.

File: Dockerfile [hunk 6]
```diff
@@ -1,41 +1,82 @@
...
+
+ARG GRAALVM_BINARY="${GRAALVM_BINARY:-graalvm-ee-java11-linux-amd64-20.1.0.1.tar.gz}"
+ENV GRAALVM_DOWNLOAD_SHA256=870e51d13e7f42df50097110b14715e765e2a726aa2609c099872995c4409d8f
+
+COPY $GRAALVM_BINARY /tmp/graalvm.tar.gz
+
+RUN set -eux \
+    && echo "Checking GraalVM hash"  \
+    && echo "${GRAALVM_DOWNLOAD_SHA256} /tmp/graalvm.tar.gz" | sha256sum --check -  \
+    && tar xzf /tmp/graalvm.tar.gz -C /opt \
```
Comment (watsh-rajneesh) [inline Dockerfile:30]: Alternatively we can do:
```
mkdir -p /opt/graalvm-ee \
&& tar xzf /tmp/graalvm.tar.gz -C /opt/graalvm-ee --strip-components=1
```

Then we dont need to do the mv in the line below.

File: Dockerfile (hunk 5 above)
Comment (sandeepbangera) [reply Dockerfile]: fixed

File: Dockerfile (hunk 6 above)
Comment (sandeepbangera) [reply Dockerfile]: fixed. Thanks for the tip

**Comment by watsh-rajneesh** [comment]
Looks good to me.

**Pull request merged**